File: AngularLovelyjubbly.Data.Sql.Models/Team.cs
namespace AngularLovelyjubbly.Data.Sql.Models
{
    public partial class Team
    {
        public Team()
        {
        }

        public int TeamId { get; set; }
        public string TeamName { get; set; }
        public string TeamNameShort { get; set; }
        public string LogoImage { get; set; }
        public string HeaderImage { get; set; }
        public string CoachImage { get; set; }
        public string CheerleaderImage { get; set; }
        //foreign key 
        public int DivisionId { get; set; }
        //navigation property
        public virtual Division Division { get; set; }
        //foreign key
        public int CoachId { get; set; }
        //navigation property
        public virtual Coach Coach { get; set; }

        public string Hex { get; set; }
        public int R { get; set; }
        public int G { get; set; }
        public int B { get; set; }
    }
}

File: AngularLovelyjubbly/ClientApp/src/app/shared/models/team.ts
import { ICoach } from './coach';
import { IDivision } from './division';

export interface ITeam {
    teamId: number;
    teamName: string;
    teamNameShort: string;
    coach: ICoach;
    division: IDivision;
    cheerleaderImage?: string;
    coachImage?: string;
    headerImage?: string;
    logoImage?: string;
    hex: string;
    r: number;
    g: number;
    b: number;
}

export class Team implements ITeam {

    constructor(public teamId: number = 0,
        public teamName: string = '',
        public teamNameShort: string = '',
        public coach: ICoach = null,
        public division: IDivision = null,
        public cheerleaderImage?: string,
        public coachImage?: string,
        public headerImage?: string,
        public logoImage?: string,
        public hex: string = '',
        public r: number = null,
        public g: number = null,
        public b: number = null) {
    }
}
